components/nimbus/src/enrollment.rs

```diff
@@ -373,40 +375,72 @@ impl ExperimentEnrollment {
                 enrollment_id,
                 branch,
                 ..
-            } => EnrollmentChangeEvent::new(
-                &self.slug,
-                enrollment_id,
-                branch,
-                None,
-                EnrollmentChangeEventType::Enrollment,
-            ),
+            } => {
```

Comment:
This function already described itself as "create a telemetry event", but this patch now has the function both call `nimbus_events::enrollment.record` *and* returns the existing `EnrollmentChangeEvent`. I'm trying to get my head around why we seem to now have 2 things, both of which seem to be broadly described as "telemetry events", and this function both emits one type and returns another (presumably for the caller to emit)?

(Maybe there's a future where the "old" style are removed?)

Reply:
Yes, I have been working on some commits locally that remove the `EnrollmentChangeEvents` that get bubbled up across the FFI, but haven't pushed them up yet. I wanted to try and keep the commits in some logical order and I was distracted tinkering with how to get the glean_parser python script into the build steps.

Reply:
That was a pretty big refactor because we end up using the events in a lot of tests to ensure things operated correctly, and that `enrollment.rs` file is getting to be nearly 4k lines long :(